src/actions/blockchain.ts

```diff
@@ -359,8 +359,11 @@ export const claimSellerFundsFromSeveral = (
         body: `Claiming ${buyName} tokens from ${sellName}-${buyName} auction. Please check ${activeProvider}`,
       },
     }))
-    const claimReceipt = await getUnclaimedSellerFundsFromSeveral(sell, buy, currentAccount, indices)
-    console.log('​claimReceipt => ', claimReceipt)
+    const [claimReceipt] = await Promise.all([
+      getUnclaimedSellerFundsFromSeveral(sell, buy, currentAccount, indices),
+      updateMainAppState(),
```

Comment:
`updateMainState` should be fired after `getUnclaimedSellerFundsFromSeveral` tx is executed and mined, otherwise it can fetch the old state. And even then it won't necessarily bring in the new state as happened in owl project (MMask is to blame?)
You don't need to `await` on `updateMainState`, you can just start it after receipt is received and it'll run it's course by itself without stalling `claimSellerFundsFromSeveral`  function execution

Reply:
good call - also yes, I checked this in the app and it doesn't update appropriately